securedrop/debian/securedrop-app-code.postinst

```diff
@@ -168,6 +168,23 @@ update_apache2_headers(){
     sed -i 's/^Header always set Cross-Origin-Resource-Policy "same-site"/Header always set Cross-Origin-Resource-Policy "same-origin"/g' "$1"
 }
 
+# Export the journalist public key out of the GPG keyring to journalist.pub
+# as part of the GPG -> Sequoia migration.
+export_journalist_public_key() {
+    # config.py is root-writable, so it's safe to run it as root to extract the fingerprint
+    journalist_pub="/var/lib/securedrop/journalist.pub"
+    if ! test -f $journalist_pub; then
+        # Based on sdconfig.py, this should work with very old config.py files.
+        fingerprint=$(cd /var/www/securedrop; python3 -c "import config; print(config.JOURNALIST_KEY)")
+        # Set up journalist.pub as root/www-data 640 before writing to it.
+        install --mode=640 --owner=root --group=www-data $journalist_pub
+        # FIXME: we should verify the exported key matches the fingerprint we wanted.
```

Comment:
I think the reason I left this as a fixme was because to do so we'd have to use gpg :| but we can just copy the `validate-gpg-key.sh` from admin/bin/.

really we should install `sq` just as a utility/debugging tool, I'll file an issue for that...

Reply:
I re-read validate-gpg-key and I don't think it's usable here because it just verifies that the file contains a key with that fingerprint, not that it _only_ contains that fingerprint.

To list out the properties we want to validate:
1. The file contains a valid PGP key.
2. The file contains one PGP key.
3. The file contains a PGP key matching the specified fingerprint.

This is pretty straightforward to do with pretty_bad_protocol actually, it's like:
`gpg.import_keys(key).fingerprints == [fingerprint]`

of course that would be evil to do in this PR, so maybe we introduce a new Sequoia function for this? hmmm :thinking: